PROJECT_HEURISTICS.md

```diff
@@ -0,0 +1,261 @@
+
+2. platform specific `bsc.exe` (formatter)
+- Starting from the current open file, walk the filesystem upwards, until `./node_modules/bs-platform` is located
+- `bsc.exe` is located at `path.join(foundDirectory, "./node_modules/bs-platform", platform, "bsc.exe")`
+
+3. `bsb` (start a build/get freshest data)
+- Starting from the current open file, walk the filesystem upwards, until `./node_modules/.bin/bsb` is located
+- `bsb` is located at `path.join(foundDirectory, "./node_modules/.bin", "bsb")`
+
+## Cases
+### Simple project
+
+```
+/folder1
+  /src
+    App.res
+  /lib
+    /bs
+      .compiler.log
+  /node_modules
+    /bs-platform
+      /darwin|linux|win32
+        bsc.exe
+    /.bin
+      bsb
+  package.json
+  bsconfig.json
+```
+
+Typical workflow:
+1. run `yarn` or `npm install` in `/folder1`
+2. build: `npx/yarn bsb` in `/folder1`
+
+When `/folder1/src/App.res` is openend in the editor:
+1. `bsc.exe` can be found in `folder1/node_modules/bs-platform/{platform}/bsc.exe`
+2. `.compiler.log` can be found in `folder1/lib/bs/.compiler.log`
+3. `bsb` to get a build for this project and get the freshest data: `folder1/node_modules/.bin/bsb`
+
+
+### Monorepo with yarn workspace with one ReScript project in subfolder
+```
+/root
+  /folder1
+    /src
+      App.res
+    /lib
+      /bs
+        .compiler.log
+    /node_modules
+      /.bin
+        bsb
+    package.json
+    bsconfig.json
+  /node_modules
+    /bs-platform
+      /darwin|linux|win32
+        bsc.exe
+    /.bin
+      bsb
+  package.json
+  yarn.lock
+```
+
+Typical workflow:
+1. run `yarn` in the root
+2. build folder1: `cd folder1 && yarn bsb`
+
+When `/folder1/src/App.res` is openend in the editor:
+1. `bsc.exe` can be found in `/root/node_modules/bs-platform/{platform}/bsc.exe`
+2. `.compiler.log` can be found in `/root/folder1/lib/bs/.compiler.log`
+3. `bsb` to get a build for this project and get the freshest data: `/root/folder1/node_modules/.bin/bsb`. Note that there's also a one in `/root/node_modules/.bin/bsb`. The bsb in the subfolder is a symlink to the one in the root's node_modules. Bsb needs a `bsconfig.json` in the current working directory, so the cwd for bsb needs to be `/root/folder1`
+
+### Yarn workspace with multiple ReScript projects in subfolders
+
+Monorepo with two subfolders `folder1` and `folder2` on the same bs-platform version.
+
+```
+/root
+  /folder1
+    /src
+      Folder1App.res
+    /lib
+      /bs
+        .compiler.log
+    /node_modules
+      /.bin
+        bsb
+    package.json
+    bsconfig.json
+  /folder2
+    /src
+      Folder2App.res
+    /lib
+      /bs
+        .compiler.log
+    /node_modules
+      /.bin
+        bsb
+    package.json
+    bsconfig.json
+  /node_modules
+    /bs-platform
+      /darwin|linux|win32
+        bsc.exe
+    /.bin
+      bsb
+  package.json
+  yarn.lock
+```
+
+Typical workflow:
+1. run `yarn` in the root
+2. build folder1: `cd folder1 && yarn bsb`
+3. build folder2: `cd folder2 && yarn bsb`
+
```

Comment:
this is incorrect. user should only run bsb in the main repo.
The project is organised with the assumption that there's only main entry point, user should never run `bsb` in different directories.